metrics/xrootd_metrics.go

```diff
@@ -1740,7 +1792,90 @@ func HandleSummaryPacket(packet []byte) error {
 	return nil
 }
 
-// handleOSSPacket processes the OSS plugin stats
+func updateIntCounter(new int, old int, counter prometheus.Counter) int {
+	incBy := math.Max(0, float64(new-old))
+	counter.Add(incBy)
+	if new < old {
+		return old
+	}
+	return new
+}
+func updateFloat64Counter(new float64, old float64, counter prometheus.Counter) float64 {
+	incBy := new - old
+	if incBy < 0 {
+		incBy = 0
+	}
+	counter.Add(incBy)
+	if new < old {
+		return old
+	}
+	return new
+}
+
+// handleS3CacheStats processes the S3 cache stats
+// It expects the blobs to be in JSON format and will update the metrics accordingly
+// It returns an error if the blobs are empty or if there is an error in unmarshalling
+// the JSON data
+// It also handles the case where the total IO or wait time is less than the previous value
+// by resetting the increment to 0.
+// The s3file_stats event schema is as follows:
+//
+//	{
+//	  "event": "s3file_stats",
+//	  "hit_b": 0,
+//	  "miss_b": 0,
+//	  "full_hit": 0,
+//	  "part_hit": 0,
+//	  "miss": 0,
+//	  "bypass_b": 12265,
```

Comment:
Since you call this a schema, I think it'd be better to indicate valid value types instead of mostly 0s, e.g.:
```
{
    "event": "s3file_stats",
    "hit_b": uint,
    "miss_b": uint,
    "bypass_s": float,
```
(I don't know if these are correct, just giving an example).